aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
+	  - {ssn:regex(\d{3}-\d{2}-\d{4})}
+	  - 123-45-6789
+	  - String must match the provided regular expression.
+	* - ``required``
+	  - {name:required}
+	  - Steve
+	  - Denotes a parameter is required; you need not specify this since it is the default.
+	* - ``any|regex|value``
+	  - {foo:a|b|c}
+	  - a
+	  - If the constraint isn't known, it is treated as a regex (if provided in the constraints dictionary - not within a template)
+
+.. tip:: To constrain a parameter to a known set of possible values, you can use a regex: ``{action:regex(list|get|create)}``. This would only match the ``action`` route value to ``list``, ``get``, or ``create``. If passed into the constraints dictionary, the string "list|get|create" would be equivalent.
+
+Constraints can be *chained*, so you can specify that a route value is of a certain type and also must fall within a certain range, like so: ``{age:int:range(1,120)}``. You can use the ``*`` character as a prefix to a route value name to bind to the rest of the URI. For example, ``blog/{*slug}`` would match any URI that started with ``/blog/`` and had any value following it (which would be assigned to the ``slug`` route value.
+
+Route templates must be unambiguous, or they will be ignored. For example, consider these two routes:
+
+1. ``{blog}/{year}/{month=1}``
+2. ``{blog}/{year}``
+
+The second route will never be used, since anything it would match will be matched by the first route (because it has a default value for month). If you wanted to construct a separate route that would for instance include all months or ask the user to select a month, you could change the route to include constant values, like this:
+
+1. ``monthly/{blog}/{year}/{month=1}``
+2. ``annual/{blog}/{year}``
+
+.. note:: There is a special case route for filenames, such that you can define a route value like ``files/{filename}.{ext?}``. When both ``filename`` and ``ext`` exist, both values will be populated. However, if only ``filename`` exists in the URL, the trailing period ``.`` is also optional. Thus, these would both match: ``/files/foo.txt`` and ``/files/foo``.
+
+.. tip:: Enable :doc:`logging` to see how the built in routing implementations, such as ``TemplateRoute``, match requests.
+
+Route Builder Extensions
+------------------------
+Several `extension methods on RouteBuilder <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteBuilderExtensions.cs>`_ are available for convenience. The most common of these is ``MapRoute``, which allows the specification of a route given a name and template, and optionally default values, constraints, and/or data tokens. When using these extensions, you must have specified the ``DefaultHandler`` and ``ServiceProvider`` properties of the ``RouteBuilder`` instance to which you're adding the route. These extensions all make use of the ``TemplateRoute`` type to add new route templates to whichever ``IRouter`` is configured to as the ``DefaultHandler``.
```

Comment:
> These extensions all make use of the ``TemplateRoute`` type to add new route templates to whichever ``IRouter`` is configured to as the ``DefaultHandler``.

These ``MapRoute`` extensions add new ``TemplateRoute`` instances to the ``RouteBulider`` that each target the ``IRouter`` configured as the ``DefaultHandler``.